crates/redeem/src/benchmarking.rs

```diff
@@ -168,238 +174,309 @@ benchmarks! {
             ..Vault::new(vault_id.clone())
         };
 
-        VaultRegistry::<T>::insert_vault(
-            &vault_id,
-            vault
-        );
+        VaultRegistry::<T>::insert_vault(&vault_id, vault);
 
-        mint_wrapped::<T>(&origin, amount);
+        mint_wrapped::<T>(&caller, amount);
 
-        assert_ok!(Oracle::<T>::_set_exchange_rate(get_collateral_currency_id::<T>(),
+        assert_ok!(Oracle::<T>::_set_exchange_rate(
+            get_collateral_currency_id::<T>(),
             UnsignedFixedPoint::<T>::one()
         ));
-    }: _(RawOrigin::Signed(origin), amount, btc_address, vault_id.clone())
 
-    liquidation_redeem {
-        assert_ok!(Oracle::<T>::_set_exchange_rate(get_collateral_currency_id::<T>(),
+        #[extrinsic_call]
+        _(RawOrigin::Signed(caller), amount, btc_address, vault_id.clone());
+    }
+
+    #[benchmark]
+    pub fn liquidation_redeem() {
+        assert_ok!(Oracle::<T>::_set_exchange_rate(
+            get_collateral_currency_id::<T>(),
             UnsignedFixedPoint::<T>::one()
         ));
 
-        let origin: T::AccountId = account("Origin", 0, 0);
+        let caller = whitelisted_caller();
         let vault_id = get_vault_id::<T>();
         let amount = 1000;
 
         register_public_key::<T>(vault_id.clone());
 
-        VaultRegistry::<T>::insert_vault(
-            &vault_id,
-            Vault::new(vault_id.clone())
-        );
+        VaultRegistry::<T>::insert_vault(&vault_id, Vault::new(vault_id.clone()));
 
-        mint_wrapped::<T>(&origin, amount.into());
+        mint_wrapped::<T>(&caller, amount.into());
 
         mint_collateral::<T>(&vault_id.account_id, 100_000u32.into());
-        assert_ok!(VaultRegistry::<T>::try_deposit_collateral(&vault_id, &collateral(100_000)));
+        assert_ok!(VaultRegistry::<T>::try_deposit_collateral(
+            &vault_id,
+            &collateral(100_000)
+        ));
 
-        assert_ok!(VaultRegistry::<T>::try_increase_to_be_issued_tokens(&vault_id, &wrapped(amount)));
+        assert_ok!(VaultRegistry::<T>::try_increase_to_be_issued_tokens(
+            &vault_id,
+            &wrapped(amount)
+        ));
         assert_ok!(VaultRegistry::<T>::issue_tokens(&vault_id, &wrapped(amount)));
 
         VaultRegistry::<T>::liquidate_vault(&vault_id).unwrap();
         let currency_pair = VaultCurrencyPair {
             collateral: get_collateral_currency_id::<T>(),
-            wrapped: get_wrapped_currency_id::<T>()
+            wrapped: get_wrapped_currency_id::<T>(),
         };
-    }: _(RawOrigin::Signed(origin), currency_pair, amount.into())
 
-    execute_redeem {
-        let origin: T::AccountId = account("Origin", 0, 0);
+        #[extrinsic_call]
+        _(RawOrigin::Signed(caller), currency_pair, amount.into());
+    }
+
+    #[benchmark]
+    pub fn execute_redeem(h: Linear<2, 10>, i: Linear<1, 10>, o: Linear<2, 3>, b: Linear<1, 1_024>) {
```

Comment:
Just a note: I _think_ that we're counting some of this factors twice. For example, adding an input will increase `i`, but it will also increase `b`. But at worst we'll overestimate the weight, which I guess is acceptable

Reply:
I had assumed the benchmarking would measure weights with respect to the relative change in each parameter. What I mean to say is for example I assume it would trial various settings of `b` when `i = x` to see how that affects computational and storage costs so the fact `i` may change the size of the Bitcoin transaction would not matter.

Reply:
It's not quite clear to me how it works: https://github.com/paritytech/substrate/blob/master/frame/benchmarking/src/analysis.rs

But worst case scneario it's just overkill, which is fine